punchcut/src/scaler.rs

```diff
@@ -0,0 +1,177 @@
+use super::{source::*, Context, Error, Hinting, NormalizedCoord, Outline, Result, Variation};
+
+use read_fonts::{
+    types::{Fixed, GlyphId, Tag},
+    TableProvider,
+};
+
+use core::{borrow::Borrow, str::FromStr};
+
+/// Builder for configuring a glyph scaler.
+pub struct ScalerBuilder<'a> {
+    context: &'a mut Context,
+    font_id: Option<u64>,
+    size: f32,
+    hint: Option<Hinting>,
+}
+
+impl<'a> ScalerBuilder<'a> {
+    /// Creates a new builder for configuring a scaler with the given context.
+    pub fn new(context: &'a mut Context) -> Self {
+        context.coords.clear();
+        context.variations.clear();
+        Self {
+            context,
+            font_id: None,
+            size: 0.0,
+            hint: None,
+        }
+    }
+
+    /// Sets a unique font identifier for hint state caching.
+    pub fn font_id(mut self, font_id: Option<u64>) -> Self {
+        self.font_id = font_id;
+        self
+    }
+
+    /// Sets the font size in pixels per em units. A size of 0.0 will disable
+    /// scaling and result in glyphs defined in font units.
+    pub fn size(mut self, size: f32) -> Self {
+        self.size = size.abs();
+        self
+    }
+
+    /// Sets the hinting mode.
+    pub fn hint(mut self, hint: Option<Hinting>) -> Self {
+        self.hint = hint;
+        self
+    }
+
+    /// Specifies a variation with a set of normalized coordinates. This will
+    /// clear any variations specified with the variations method.
```

Comment:
```suggestion
    /// Specifies a variation with a set of normalized coordinates.
    ///
    /// This will clear any variations specified with the variations method.
```